processing/src/main/java/io/druid/query/topn/TopNParams.java

```diff
@@ -19,37 +19,43 @@
 
 package io.druid.query.topn;
 
+import io.druid.query.ColumnSelectorPlus;
+import io.druid.query.topn.types.TopNColumnSelectorStrategy;
 import io.druid.segment.Cursor;
-import io.druid.segment.DimensionSelector;
 
 /**
  */
 public class TopNParams
 {
-  private final DimensionSelector dimSelector;
   private final Cursor cursor;
   private final int cardinality;
   private final int numValuesPerPass;
+  private final ColumnSelectorPlus<TopNColumnSelectorStrategy> selectorPlus;
 
   protected TopNParams(
-      DimensionSelector dimSelector,
+      ColumnSelectorPlus<TopNColumnSelectorStrategy> selectorPlus,
       Cursor cursor,
       int numValuesPerPass
   )
   {
-    this.dimSelector = dimSelector;
+    this.selectorPlus = selectorPlus;
     this.cursor = cursor;
-    this.cardinality = dimSelector.getValueCardinality();
+    this.cardinality = selectorPlus.getColumnSelectorStrategy().getCardinality(selectorPlus.getSelector());
     this.numValuesPerPass = numValuesPerPass;
 
     if (cardinality < 0) {
       throw new UnsupportedOperationException("Cannot operate on a dimension without a dictionary");
     }
   }
 
-  public DimensionSelector getDimSelector()
+  public Object getDimSelector()
```

Comment:
cool, I was originally intending for that to be the method of getting a selector but getSelectorPlus() overrides that, changed this back to `DimensionSelector` added a comment about that